Admin/Get-MigrationReport.ps1

```diff
@@ -0,0 +1,181 @@
+    New-Item $File -Type file -Force
+    "This Move Request has the following infomration:" >> ($File)
+    "-----------------------------------------------------------------------" >> ($File)
+    "the status of this Move Request is " + $MoveRequestStatistics.Status.tostring() + " with " + $MoveRequestStatistics.Status + " Percent"  >> ($File)
+    "" >> ($File)
+    $MoveRequestStatistics.Message.ToString() >> ($File)
+    "" >> ($File)
+    "-----------------------------------------------------------------------" >> ($File)
+    "" >> ($File)
+    "The Move Request has the following Failures:" >> ($File)
+    $MoveRequestStatistics.Report.Failures | Group-Object  FailureType | Format-Table Count, Name >> ($File)
+    "-----------------------------------------------------------------------" >> ($File)
+    "" >> ($File)
+    "Here is more details about each Failure (Note that only the last error is selected in more details):" >> ($File)
+    "" >> ($File)
+    $detailedFailure >> ($File)
+    Add-Content -Path $logFile -Value "[INFO] the summary report has been created successfully."
+}
+
+#===================MAIN======================
+New-Item $OutputFolder -ItemType Directory -Force | Out-Null
+New-Item $logFile -Type File -Force -ErrorAction SilentlyContinue | Out-Null
+
+foreach ($Mailbox in $Identity) {
+
+    $ErrorActionPreference = 'SilentlyContinue'
+    $MoveRequest = Get-MoveRequest $Mailbox -ErrorAction SilentlyContinue
+    $MoveRequestStatistics = Get-MoveRequestStatistics $Mailbox -IncludeReport -DiagnosticInfo "showtimeslots, showtimeline, verbose" -ErrorAction SilentlyContinue
+    if ($null -eq $MoveRequest) {
+        Write-Host -ForegroundColor Red -Value "[ERROR] The MoveRequest for the $Mailbox cannot be found, please check spelling and try again!"
+        Add-Content -Path $logFile -Value "[ERROR] The MoveRequest for the $Mailbox cannot be found, please check spelling and try again!"
+    }
+    $Batch = $MoveRequestStatistics.BatchName
+    $MigrationBatch = Get-MigrationBatch $Batch -IncludeReport -DiagnosticInfo "showtimeslots, showtimeline, verbose" -ErrorAction SilentlyContinue
+    $UserMigration = Get-MigrationUser $Mailbox  -ErrorAction SilentlyContinue
+    $UserMigrationStatistics = Get-MigrationUserStatistics $Mailbox -IncludeSkippedItems -IncludeReport -DiagnosticInfo "showtimeslots, showtimeline, verbose" -ErrorAction SilentlyContinue
+    $Endpoint = $MigrationBatch.SourceEndpoint
+    $MigrationEndPoint = Get-MigrationEndpoint -Identity $Endpoint -DiagnosticInfo Verbose -ErrorAction SilentlyContinue
+    $MailboxStatistics = Get-MailboxStatistics $Mailbox -IncludeMoveReport -IncludeMoveHistory -ErrorAction SilentlyContinue
+    $MoveHistory = Get-MailboxStatistics $Mailbox -IncludeMoveReport -IncludeMoveHistory -ErrorAction SilentlyContinue
```

Comment:
Why are we running the same cmdlet here and storing to two different variables? This has zero benefit.